beacon_node/beacon_chain/src/attester_per_slot_service.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{BeaconChain, BeaconChainTypes};
+use slog::{debug, error};
+use slot_clock::SlotClock;
+use std::sync::Arc;
+use task_executor::TaskExecutor;
+use tokio::time::sleep;
+use types::{BeaconState, RelativeEpoch, Slot};
+
+/// Spawns a routine which produces an unaggregated attestation at every slot.
+///
+/// This routine will run once per slot
+pub fn start_attester_per_slot_service<T: BeaconChainTypes>(
+    executor: TaskExecutor,
+    chain: Arc<BeaconChain<T>>,
+) {
+    // TODO: AI(joel) Only run the service if validator monitor is enabled
+    // Paul has made a refacto of that bit in another PR, I will rebase
+    // once it's merged
+    executor.clone().spawn(
+        async move { attester_per_slot_service(executor, chain).await },
+        "attester_per_slot_service",
+    );
+}
+
+/// Loop indefinitely, calling `BeaconChain::produce_unaggregated_attestation` every 4s into each slot.
+async fn attester_per_slot_service<T: BeaconChainTypes>(
+    executor: TaskExecutor,
+    chain: Arc<BeaconChain<T>>,
+) {
+    let slot_duration = chain.slot_clock.slot_duration();
+    let additional_delay = slot_duration / 3;
+
+    loop {
+        match chain.slot_clock.duration_to_next_slot() {
+            Some(duration) => {
+                sleep(duration + additional_delay).await;
+
+                debug!(
+                    chain.log,
+                    "Produce an unaggregated attestation";
+                );
+
+                // I am not sure the impact of cloning the state at every epoch yet
```

Comment:
Cloning the head state is highly memory intensive (100s of MBs), so it's definitely something we want to avoid. I make a suggestion in a later comment that should remove the need for it 👍

Reply:
Just realised it's an Arc so we should be okay cloning it as it won't clone the underlying type?